lib/alchemy/upgrader/four_point_zero.rb

```diff
@@ -1,3 +1,118 @@
+require 'thor'
+
+class Alchemy::Upgrader::FourPointZeroTasks < Thor
+  include Thor::Actions
+
+  no_tasks do
+    def convert_available_contents
+      config = read_config
+
+      elements_with_available_contents, new_elements = config.partition do |e|
+        e['available_contents']
+      end
+
+      return if elements_with_available_contents.empty?
+
+      print 'Converting to `nestable_elements` ... '
+      elements_with_available_contents.inject(new_elements) do |ne, old_element|
+        ne << modify_old_element(old_element.dup)
+        build_new_elements(old_element).inject(ne) do |ne, element_from_contents|
+          ne << element_from_contents
+        end
+      end
+      new_elements = new_elements.uniq.sort {|a, b| a['name'] <=> b['name'] }
+      puts 'done.'
+
+      print 'Writing new `config/alchemy/elements.yml` ... '
+      write_config(new_elements)
+      puts 'done.'
+
+      print 'Removing `render_new_content_link` helper from editor partials ... '
+      remove_new_content_link_from_editor_partials
+      puts 'done.'
+
+      puts 'Adding hints for rendering nested elements to your views: '
+      add_render_nested_elements_hints
+    end
+  end
+
+  private
+
+  def backup_config
+    print "Copying existing config file to config/alchemy/elements.yml.old ... "
+    FileUtils.copy  Rails.root.join('config', 'alchemy', 'elements.yml'),
+                    Rails.root.join('config', 'alchemy', 'elements.yml.old')
```

Comment:
Style/AlignParameters: Align the parameters of a method call if they span more than one line.